src/magentic_ui/teams/orchestrator/_orchestrator.py

```
@@ -985,112 +1016,328 @@ async def _orchestrate_step_execution(
         ):
+            print(f"{GREEN}Preparing final answer...{RESET}")
             await self._prepare_final_answer("Max rounds reached.", cancellation_token)
             return
 
         self._state.n_rounds += 1
-        context = self._thread_to_context()
-        # Update the progress ledger
 
-        progress_ledger_prompt = self._get_progress_ledger_prompt(
-            self._state.task,
-            self._state.plan_str,
-            self._state.current_step_idx,
-            self._team_description,
-            self._agent_execution_names,
-        )
+        # Gets the current step being executed
+        current_step = self._state.plan[self._state.current_step_idx]
+        print(f"{BLUE}Current Step: {current_step}{RESET}")
 
-        context.append(UserMessage(content=progress_ledger_prompt, source=self._name))
+        # ╭────────────────────────────────────────────────────────────────────────────╮
+        # │  IF SENTINEL TASKS FLAG ARE ACTIVE AND STEP IS A SENTINEL STEP             │
+        # ╰────────────────────────────────────────────────────────────────────────────╯
+        if self._config.sentinel_tasks and isinstance(current_step, SentinelPlanStep):
+            # Logs the start of the sentinel step
+            await self._log_message_agentchat(
+                f"Executing sentinel step: {current_step.title}",
+                metadata={"internal": "no", "type": "sentinel_start"},
+            )
 
-        progress_ledger = await self._get_json_response(
-            context, self._validate_ledger_json, cancellation_token
-        )
-        if self._state.is_paused:
-            await self._request_next_speaker(self._user_agent_topic, cancellation_token)
-            return
-        assert progress_ledger is not None
-        # log the progress ledger
-        await self._log_message_agentchat(dict_to_str(progress_ledger), internal=True)
+            # Execute the sentinel step
+            try:
```

this try/catch here should likely be avoided and let the program crash, instead we can make sure execute_sentinel_step can recover from failures if needed, otherwise just letting the program crash is as good as other options